### PR TITLE
PAT integration with ui and apis

### PR DESCRIPTION
**Pull request by tapishr**
Integrated mine and @MuskanPaliwal 's work on PAT and got them to work together
- [x] Manually Tested
- [x] Build Succeding 

## Summary by CodeRabbit

- **New Features**
	- Enhanced setup process to include the removal of previous installations for better data integrity.
	- Introduced a new API endpoint for retrieving user repositories by topic and provider.

- **Bug Fixes**
	- Added error handling in authentication to improve reliability and user experience.

- **Refactor**
	- Updated database utility functions to support new features and improve error handling.

### CODE REVIEW

File: pages/api/dpu/setup.ts [hunk 1]
```diff
@@ -11,6 +9,17 @@ const setupHandler = async (req: NextApiRequest, res: NextApiResponse) => {
...
+	let installsRemoved = true;
+	await removePreviousInstallations(jsonBody.installationId).catch(err => {
+		console.error(`[setupHandler] Unable to remove previous installations for ${jsonBody.installationId}`, err);
+		res.status(500).json({"error": "Internal Server Error"});
+		installsRemoved = false;
+		return;
+	});
+	if (!installsRemoved) {
+		return;
+	}
```
Comment (avikalpg) [inline pages/api/dpu/setup.ts:22]: Is this actually working? The `res.status` command is inside the catch block (not the scope of the handler function) and I am not very confident that line 20 will consistently work.

If you awaiting, then why not directly use a `try-catch` block? That way, you will not even need the `installsRemoved` variable. If the async function fails, you can directly return the 500 response.

File: utils/db/repos.ts [hunk 2]
```diff
@@ -38,6 +38,25 @@ export const getRepos = async (allRepos: RepoIdentifier[]) => {
...
+	const repos: RepoIdentifier[] = await conn.query(getRepoQuery)
+		.then((dbResponse) => {
+			return dbResponse.rows.map((row) => ({
+				repo_name: row.repo_name,
+				repo_owner: row.repo_owner,
+				repo_provider: row.repo_provider,
+			}));
+		})
+		.catch((err: Error) => {
+			console.error(`[db/getUserRepositoriesByTopic] Could not get repos for topic id ${topicId}`, err);
```
Comment (avikalpg) [inline utils/db/repos.ts:54]: Low LOGAF.

Shouldn't we also log the query that failed?

**Review by avikalpg — CHANGES_REQUESTED** [review]
Just convert the .catch to try-catch so that you can eliminate the multiple handling of thrown error. Everything else is fine.

**Review by MuskanPaliwal — APPROVED** [review]

**Review by avikalpg — APPROVED** [review]

File: pages/api/dpu/repos.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        || Array.isArray(topicId) || Array.isArray(provider)
+        || topicId.length === 0 || provider.length === 0) {
+        res.status(400).json({ error: 'Invalid get request body' });
+		return;
+    }
+    const repoList: RepoIdentifier[] = await getUserRepositoriesByTopic(topicId, provider);
+	if (repoList.length == 0) {
+		return res.status(500).json({error: 'Unable to get repo list'});
+	}
+	return res.status(200).json({repoList: repoList});
```
[the remark this answers is not in the document]
Comment (tapishr) [reply pages/api/dpu/repos.ts]: Done. Removed length condition, now empty list is treated as list

**Review by avikalpg — APPROVED** [review]

**Review by avikalpg — APPROVED** [review]

**Pull request merged**